x-pack/plugins/enterprise_search/public/applications/workplace_search/views/content_sources/components/add_source/add_source_logic.test.ts

```diff
@@ -353,6 +358,30 @@ describe('AddSourceLogic', () => {
         expect(navigateToUrl).toHaveBeenCalledWith(`${ADD_GITHUB_PATH}/configure${queryString}`);
       });
 
+      describe('Github error edge case', () => {
+        it('handles "organization" redirect and displays error', () => {
+          const githubQueryString =
+            '?error=redirect_uri_mismatch&error_description=The+redirect_uri+MUST+match+the+registered+callback+URL+for+this+application.&error_uri=https%3A%2F%2Fdocs.github.com%2Fapps%2Fmanaging-oauth-apps%2Ftroubleshooting-authorization-request-errors%2F%23redirect-uri-mismatch&state=%7B%22action%22%3A%22create%22%2C%22context%22%3A%22organization%22%2C%22service_type%22%3A%22github%22%2C%22csrf_token%22%3A%22TTAyYktJSkU5Um9PQ0p5clNEbGk0MnM0cXhqMXZXaDBxdG8zeXQ3YkRnYkIyempWaUMwMGpuZmpLR25pcHRKT0RxUXFvbTJGM20vZE5RcUpEcVhWNEE9PQ%3D%3D%22%2C%22index_permissions%22%3Afalse%7D';
+          AddSourceLogic.actions.saveSourceParams(githubQueryString);
+
+          expect(navigateToUrl).toHaveBeenCalledWith('/');
+          expect(setErrorMessage).toHaveBeenCalledWith(
+            'The redirect_uri MUST match the registered callback URL for this application.'
+          );
+        });
+
+        it('handles "account" redirect and displays error', () => {
+          const githubQueryString =
+            '?error=redirect_uri_mismatch&error_description=The+redirect_uri+MUST+match+the+registered+callback+URL+for+this+application.&error_uri=https%3A%2F%2Fdocs.github.com%2Fapps%2Fmanaging-oauth-apps%2Ftroubleshooting-authorization-request-errors%2F%23redirect-uri-mismatch&state=%7B%22action%22%3A%22create%22%2C%22context%22%3A%22account%22%2C%22service_type%22%3A%22github%22%2C%22csrf_token%22%3A%22TTAyYktJSkU5Um9PQ0p5clNEbGk0MnM0cXhqMXZXaDBxdG8zeXQ3YkRnYkIyempWaUMwMGpuZmpLR25pcHRKT0RxUXFvbTJGM20vZE5RcUpEcVhWNEE9PQ%3D%3D%22%2C%22index_permissions%22%3Afalse%7D';
+          AddSourceLogic.actions.saveSourceParams(githubQueryString);
+
+          expect(navigateToUrl).toHaveBeenCalledWith(PERSONAL_SOURCES_PATH);
+          expect(setErrorMessage).toHaveBeenCalledWith(
+            'The redirect_uri MUST match the registered callback URL for this application.'
```

Comment:
**Optional, non-blocking:**
I understand that this is an edge case, and might not worth the time investment, so I'm good with merging as-is.
 
For the personal dashboard maybe we could offer a friendlier message?
Like: `Could not connect the source, reach out to your admin for help. Error message: "The redirect_uri MUST match..."`. The first part could be i18n-ed.

Reply:
As James mentioned [here](https://github.com/elastic/workplace-search-team/issues/1850#issuecomment-878697148), there are potentially other cases that could trigger an error so that string would not always be true.

Reply:
Fixed:
![image](https://user-images.githubusercontent.com/1869731/125498055-73a93573-7b99-4e12-86b6-565a13a321e1.png)

https://github.com/elastic/kibana/pull/105302/commits/64751ea625c25a50be602d1d65263113777d992b